File: 2022-2023/sourcecode/lab/lab11/FoodyJava/app/src/main/java/pl/udu/uwr/pum/foodyjava/adapters/favorite/FavoriteViewHolder.java
package pl.udu.uwr.pum.foodyjava.adapters.favorite;

import androidx.navigation.NavDirections;
import androidx.navigation.Navigation;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;

import pl.udu.uwr.pum.foodyjava.data.Meal;
import pl.udu.uwr.pum.foodyjava.databinding.ListItemRvBinding;
import pl.udu.uwr.pum.foodyjava.ui.fragments.FavoriteFragmentDirections;
import pl.udu.uwr.pum.foodyjava.ui.fragments.MealListFragmentDirections;

public class FavoriteViewHolder extends RecyclerView.ViewHolder {
    private final ListItemRvBinding binding;
    public FavoriteViewHolder(ListItemRvBinding binding) {
        super(binding.getRoot());
        this.binding = binding;
    }

    public void bind(Meal item){
        binding.name.setText(item.strMeal);
        Glide.with(binding.getRoot())
                .load(item.strMealThumb)
                .into(binding.image);
        binding.getRoot().setOnClickListener(view -> {
            NavDirections action = FavoriteFragmentDirections
                    .actionFavoriteFragmentToMealDetailFragment(
                            item.idMeal
                    );
            Navigation.findNavController(binding.getRoot()).navigate(action);
        });
    }
}

File: 2022-2023/sourcecode/lab/lab11/Foody/app/src/main/java/pl/udu/uwr/pum/foody/adapters/favorite/FavoriteViewHolder.kt
package pl.udu.uwr.pum.foody.adapters.favorite

import androidx.navigation.NavDirections
import androidx.navigation.Navigation
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import pl.udu.uwr.pum.foody.data.Meal
import pl.udu.uwr.pum.foody.databinding.ListItemRvBinding
import pl.udu.uwr.pum.foody.ui.fragments.FavoriteFragmentDirections
import pl.udu.uwr.pum.foody.ui.fragments.FoodListFragmentDirections

class FavoriteViewHolder(private val binding: ListItemRvBinding)
    : RecyclerView.ViewHolder(binding.root){
        fun bind(item: Meal){
            binding.name.text = item.strMeal
            Glide.with(binding.root)
                .load(item.strMealThumb)
                .into(binding.image)
            binding.root.setOnClickListener {
                val action: NavDirections = FavoriteFragmentDirections
                    .actionFavoriteFragmentToFoodDetailFragment(
                        item.idMeal
                    )
                Navigation.findNavController(binding.root).navigate(action)
            }
        }
}